libs/zedUpload/datastore.go

```diff
@@ -262,20 +262,26 @@ func NewDronaCtx(name string, noHandlers int) (*DronaCtx, error) {
 	return &dSync, nil
 }
 
+func reqPostSize(req *DronaRequest, dronaCtx *DronaCtx, stats types.UpdateStats) {
+	req.doneParts = stats.DoneParts
+	dronaCtx.postSize(req, stats.Size, stats.Asize)
+}
+
 func statsUpdater(req *DronaRequest, dronaCtx *DronaCtx, prgNotif types.StatsNotifChan) {
 	ticker := time.NewTicker(StatsUpdateTicker)
 	defer ticker.Stop()
-	var stats types.UpdateStats
+	var newStats, stats types.UpdateStats
 	var ok bool
 	for {
 		select {
-		case stats, ok = <-prgNotif:
+		case newStats, ok = <-prgNotif:
```

Comment:
Can you please add a comment why exactly we use a 'newStats' variable?
Actually this is a different bug fix (race fix) and ideally would be great a separate commit for that, but if you write an explicit comment and also mention the fix for the race in the commit message it seems would be sufficient.

Reply:
here it is: https://github.com/lf-edge/eve/pull/3236/commits/3653d4cd92e0194e00f005a452ca28a3d6f1ea5d